tiledb/sm/query/query_condition.cc

```diff
@@ -647,6 +647,15 @@ Status QueryCondition::apply_clause(
   const bool var_size = attribute->var_size();
   const bool nullable = attribute->nullable();
   switch (attribute->type()) {
+    case Datatype::BYTE:
+      return apply_clause<std::byte>(
```

Comment:
Unfortunately, the constructor for `QueryCondition` ignores types and thus simply assumes that all types have comparison operators. If `BYTE` has no less-than operator, this should return an error.